### PR TITLE
RUMM-322 Add Ktx module

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android-ktx/src/test/kotlin/com/datadog/android/ktx/tracing/SpanExtTest.kt [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+        assertThrows<Throwable> {
+            withinSpan(operationName, mockParentSpan) {
+                lambdaCalled = true
+                throw throwable
+            }
+        }
+
+        assertThat(lambdaCalled).isTrue()
+        verify(mockSpan).finish()
```
Comment (mariusc83) [inline dd-sdk-android-ktx/src/test/kotlin/com/datadog/android/ktx/tracing/SpanExtTest.kt:186]: you can check also that mockSpan.setError was called

File: dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/AndroidTracer.kt [hunk 2]
```diff
@@ -163,6 +167,9 @@ class AndroidTracer internal constructor(
         fun logErrorMessage(span: Span, message: String) {
             val fieldsMap = mapOf(Fields.MESSAGE to message)
             span.log(fieldsMap)
+            if (span is DDSpan) {
+                span.setError(true)
```
Comment (mariusc83) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/AndroidTracer.kt:171]: You do not need this....it is already done in the AndroidLogsHandler by calling : `span.setErrorMeta` which inside calls `span.setError(true)`

File: dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/AndroidTracer.kt [hunk 3]
```diff
@@ -151,6 +152,9 @@ class AndroidTracer internal constructor(
         fun logThrowable(span: Span, throwable: Throwable) {
             val fieldsMap = mapOf(Fields.ERROR_OBJECT to throwable)
             span.log(fieldsMap)
+            if (span is DDSpan) {
+                span.setError(true)
```
Comment (mariusc83) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/AndroidTracer.kt:156]: Same here...you don't need this call

File: dd-sdk-android-ktx/src/main/kotlin/com/datadog/android/ktx/Datadog.kt [hunk 4]
```diff
@@ -0,0 +1 @@
+package com.datadog.android.ktx
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply dd-sdk-android-ktx/src/main/kotlin/com/datadog/android/ktx/Datadog.kt:1]: Oh yeah this was not meant to be committed

**Pull request merged**